vdr/doc/merge.go

```diff
@@ -0,0 +1,209 @@
+/*
+ * Nuts node
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package doc
+
+import (
+	"errors"
+	"sort"
+	"strings"
+
+	ssi "github.com/nuts-foundation/go-did"
+	"github.com/nuts-foundation/go-did/did"
+)
+
+// ErrDiffID is returned when a merge is done on documents that do not share the same ID
+var ErrDiffID = errors.New("no matching IDs")
+
+// MergeConflicted merges two DID Documents that share the same ID
+func MergeConflicted(docA did.Document, docB did.Document) (*did.Document, error) {
```

Comment:
I'd not call the func "Conflicted" because that dictates the usage/context of this function. Isn't it simply `MergeDocuments`? (also; we don't have other document merges so there won't be any confusion)

It's like reading a byte stream which happens to be from disk and calling it `ReadBytesFromDisk(stream)`